pkg/alertmanager/amcfg.go

```diff
@@ -128,6 +128,56 @@ func validateConfigInputs(am *monitoringv1.Alertmanager) error {
 	return nil
 }
 
+func (cg *configGenerator) generateGlobalConfig(
+	ctx context.Context,
+	baseConfig *alertmanagerConfig,
+	am *monitoringv1.Alertmanager,
+	amConfigs map[string]*monitoringv1alpha1.AlertmanagerConfig,
+) error {
+	if am.Spec.GlobalAlertmanagerConfig == nil {
+		return nil
+	}
+
+	// CrKey is the global alertmanager config's crkey
+	crKey := types.NamespacedName{
+		Name:      am.Spec.GlobalAlertmanagerConfig.Name,
+		Namespace: am.Namespace,
+	}
+	// Find the global alertmanagerconfig by <namespace/name> key
+	if amConfig, ok := amConfigs[crKey.String()]; ok {
+		// Add inhibitRules to baseConfig.InhibitRules without enforce namespace
+		for _, inhibitRule := range amConfig.Spec.InhibitRules {
+			baseConfig.InhibitRules = append(baseConfig.InhibitRules, cg.convertInhibitRule(&inhibitRule))
+		}
+
+		// Add routes to baseConfig.Route.Routes without enforce namespace
+		baseConfig.Route.Routes = []*route{
+			cg.convertRoute(amConfig.Spec.Route, crKey),
+		}
+
+		for _, receiver := range amConfig.Spec.Receivers {
+			receivers, err := cg.convertReceiver(ctx, &receiver, crKey)
+			if err != nil {
+				return errors.Wrapf(err, "AlertmanagerConfig %s", crKey.String())
+			}
+			baseConfig.Receivers = append(baseConfig.Receivers, receivers)
+		}
+
+		for _, muteTimeInterval := range amConfig.Spec.MuteTimeIntervals {
+			mti, err := convertMuteTimeInterval(&muteTimeInterval, crKey)
+			if err != nil {
+				return errors.Wrapf(err, "AlertmanagerConfig %s", crKey.String())
+			}
+			baseConfig.MuteTimeIntervals = append(baseConfig.MuteTimeIntervals, mti)
+		}
+
+		// Remove the global alertmanager config to avoid generating again
+		delete(amConfigs, crKey.String())
+	}
```

Comment:
I think we should have some handling for the `!ok` case here also. Perhaps simply logging the issue is ok, but if I have configured the fields and the lookup fails, I would expect to be informed.